### PR TITLE
update to ocaml 4.14, jsoo 4.0.0, ppxlib 0.26.0

### PR DESCRIPTION
**Pull request by ulugbekna**

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -24,15 +24,17 @@
 (package
  (name vscode-ocaml-platform)
  (synopsis "Visual Studio Code extension for OCaml")
+ (allow_empty) ; without this line, dune complains "The package vscode-ocaml-platform does not have any user defined
```
Comment (rgrinberg) [inline dune-project:27]: could you wrap the comment at 80 chars.

File: src/dune [hunk 2]
```diff
@@ -7,4 +7,4 @@
    ppx_tools)
  (modes js)
  (js_of_ocaml
-  (flags --source-map --pretty)))
+  (flags --source-map)))
```
Comment (rgrinberg) [inline src/dune]: Why this change?

Comment (ulugbekna) [reply src/dune]: Sorry, this was a confusion after upgrading to dune 3. I reverted that already. 
Related: https://github.com/ocaml/dune/issues/5563

**Comment by rgrinberg** [comment]
Can we separate the upgrades to separate PR's? Or at least merge them in sequence to make sure CI likes them separately. Or are they all necessarily coupled?

**Comment by ulugbekna** [comment]
> Can we separate the upgrades to separate PR's? Or at least merge them in sequence to make sure CI likes them separately. Or are they all necessarily coupled?

I'll try
